### ISSUE TITLE
Sign all Weblate's Git commits with a GPG key

### ISSUE THREAD

**Issue by SunilMohanAdapa** [WeblateOrg/weblate#1804]
Some projects (such as FreedomBox) sign all of their Git commits with a GPG key and automate the checking of these signatures. This is to ensure that:
1) Users pulling the code can trust the commits coming from known developers.
2) Added security that hosted services such as Github don't make changes to existing commits or new commits.

It would be nice if Weblate can sign its commits so that it becomes easier of such projects to verify the GPG signature after the commit is pushed to the repository.

In case of hosted Weblate, this becomes even more important. The public key used by Weblate can be publicly announced on the website and signed by Weblated developers.

**Comment by nijel** [comment]
Adding support to sign all commits automatically would be pretty simple in the [vcs code](https://github.com/WeblateOrg/weblate/blob/c45dddba1f723bd1c705f61753617bcb5408e10f/weblate/trans/vcs.py).

There are two options:

* Pass `--gpg-sign=<keyid>` to each git commit call
* Configure `commit.gpgSign` in the `global_setup` method

The feature should be probably opt-in - enabled only once GPG key is configured in the settings.

**Comment by nijel** [comment]
Thank you for your report, the issue you have reported has just been fixed. 

* In case you see problem with the fix, please comment on this issue. 
* In case you see similar problem, please open separate issue.
* If you are happy with the outcome, consider supporting Weblate by [donating](https://weblate.org/donate/).

**Comment by nijel** [comment]
Now deployed on Hosted Weblate as well, for example https://github.com/WeblateOrg/weblate/commit/b4c5b285b2a960d41fca7cff36da793f212355a1 is signed commit.
